addons/mail/static/src/chatter/web/form_renderer.scss

```diff
@@ -35,6 +35,10 @@
         --formView-sheetBg-padding-right: #{map-get($spacers, 2)};
     }
 }
+// Prevent reducing the spacing for form in dialog
+div.modal-dialog .o_form_sheet_bg {
+    padding-right: var(--formView-sheetBg-padding-x) !important;
```

Comment:
Is this rules still necessary? When I disable it on my inspector, I don't have any horizontal scroll bar. I also don't have any horizontal scrollbar on the 17.2 and master branch. Is there anything specific to do?

Reply:
After investigating 🔎 It seems that there is a chatter element in the form dialog as well. As a result, the selector applies to the dialog and cause the overflow we have.

See:

```css
.o_form_sheet_bg:has(+ .o-mail-Form-chatter.o-aside) {
    @media (max-width: 1920px) {
        --formView-sheetBg-padding-right: #{map-get($spacers, 2)};
    }
}
```

We can either add a new rule such as:
```css
.modal .o_form_sheet_bg {
    --formView-sheetBg-padding-right: --formView-sheetBg-padding-x;
}
```

Or, undo this commit: https://github.com/odoo/odoo/commit/6be1a42b27a782ab06b1bbae1ed1ea81b55741cd ?

Hello @mano-odoo :wave: Is the spacing reduction tweak necessary? I feel like it adds complexity to the styling and can have side effects elsewhere. Shouldn't we rather use `flex-gap` on the container containing the form sheet and the chatter?

Reply:
Hello @julien-banken  👋 , yes the tweak is necessary as it was the main goal of [task-3675376](https://www.odoo.com/web#id=3675376&cids=1&menu_id=4720&action=333&active_id=587&model=project.task&view_type=form) to dynamically adapt the spacing regarding the viewport to make the form view wider. Additionnaly minimizing the possibility of a horizontal scroll inside the chatter due to the o-mail-Chatter-topbar which can get quite long when all the buttons are displayed inside. 

Why do we keep a node with a `display: none;` chatter element rendered inside the modal? The rule is failing because of this. I would suggest not rendering this element if possible. Else, adapting the rule in `form_renderer.scss` to account for this corner case. (to be tested)

```scss 
// Reduce horizontal spacing if the next sibling is an aside chatter
.o_content > .o_form_renderer > .o_form_sheet_bg:has(+ .o-mail-Form-chatter.o-aside) {
    @media (max-width: 1920px) {
        --formView-sheetBg-padding-right: #{map-get($spacers, 2)};
    }
}
```
I'm not against using a flex-gap approach as you mentioned but I don't think it'll be easy to implement neither, since we currently rely on padding in the form_view you will surely have to adapt many components to fit this new approach while still making it dynamic for the aside chatter.